src/internal/kopia/conn.go

```diff
@@ -230,6 +236,17 @@ func blobStoreByProvider(
 	}
 }
 
+func configHashByProvider(ctx context.Context, s storage.Storage) (string, error) {
```

Comment:
This makes sense @ryanfkeepers .
I intended to make the hash out of account and strorage details both. Which is not accomplished here. 
I think the better place would be `pkg/repository/repository.go` as both those details are accessible here. 
Hash would be created and passed to kopia while 'Initialize', 'Connect' and 'UpdatePassword' methods called.
But this will lead to changes in lot of files (mostly test files). Do you see any concerns?

Reply:
> I intended to make the hash out of account and strorage details both. Which is not accomplished here.

An alternative suggestion:
* add a Hashing func to the `str` package for consistency.
* have Storage and Account independently own a `Hashed` function (as described in the prior comment with ConfigHash())
* Let repository make a hash out of the concatenated Storage and Account hashes (or just concat the hashes without re-hashing) for the final file name.

This ensures that: 1/ storage and account clearly own their hashing behavior, and can build expectations around the handling there as described in the prior comment, and 2/ the production of a combined hash file in the repository is a decoupled implementation of that behavior.

Reply:
> This ensures that: 1/ storage and account clearly own their hashing behavior, and can build expectations around the handling there as described in the prior comment, and 2/ the production of a combined hash file in the repository is a decoupled implementation of that behavior.

Yeah this should keep things in their own domains.

Reply:
> I'm not sure I grok the context. Can you please give me some specific examples?

if we implement hash generation in `pkg/repository/repository.go` and pass the hashed string as parameter to `Initialize`, `Connect` and `UpdatePassword` methods that would mean the function signature has to change [from `Connect(ctx context.Context, opts repository.Options) error` to `Connect(ctx context.Context, opts repository.Options, hashStr string) error`] and that would lead to change in many places primarily test files. 

For example https://github.com/alcionai/corso/blob/ec1afa8c8489fded4bf074bc719a110ff62287a7/src/internal/kopia/model_store_test.go#L881